src/WritePlasterContent.ps1

```diff
@@ -0,0 +1,246 @@
+        include: File, TemplateFile, Message, Modify, newModuleManifest, requireModule
+
+    .EXAMPLE
+        TBD
+    #>
+    [cmdletbinding()]
+    Param (
+        [Parameter(ParameterSetName = "default", Mandatory = $true, Position = 0)]
+        [ValidateSet('File','TemplateFile','Message','Modify','newModuleManifest','requireModule')]
+        [string]$Type,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Condition,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Source,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Destination,
+
+        [Parameter(ParameterSetName = "default")]
+        [switch]$OpenInEditor,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Encoding,
+
+        [Parameter(ParameterSetName = "default")]
+        [switch]$noNewLine,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Replace,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$ReplaceCondition,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Original,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$OriginalExpand,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Substitute,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$SubstituteExpand,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Path,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$rootmodule,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Author,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$CompanyName,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Description,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$ModuleVersion,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Copyright,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$ProjectUri,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$LicenseUri,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$IconUri,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Tags,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Name,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$MinimumVersion,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$MaximumVersion,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$RequiredVersion,
+
+        [Parameter(ParameterSetName = "default")]
+        [string]$Message,
+
+        [Parameter(ParameterSetName = "pipeline", ValueFromPipeLine = $true, Position = 0)]
+        [Hashtable]$obj
+    )
+
+    Process {
+        # If a hash is passed then recall this function with the hash splatted instead.
+        if ($obj -ne $null) {
```

Comment:
To shut up a PSScriptAnalyzer warning, change to `$null -ne $obj`. BTW I recommend you use Visual Studio Code to do Plaster development as it will mark the above line with a PSSA warning.